patches/minecraft/net/minecraft/tileentity/TileEntityFurnace.java.patch

```diff
@@ -62,19 +62,18 @@
              }
  
              if (itemstack.func_77973_b() == Item.func_150898_a(Blocks.field_150360_v) && itemstack.func_77960_j() == 1 && !((ItemStack)this.field_145957_n.get(1)).func_190926_b() && ((ItemStack)this.field_145957_n.get(1)).func_77973_b() == Items.field_151133_ar)
-@@ -316,6 +316,11 @@
+@@ -315,6 +315,10 @@
+         }
          else
          {
-             Item item = p_145952_0_.func_77973_b();
-+            if (!item.getRegistryName().func_110624_b().equals("minecraft"))
++            if(net.minecraftforge.fml.common.registry.GameRegistry.isFuelHandled(p_145952_0_))
```

Comment:
I think I'd prefer you did this without calling everything twice. 
Just get an `OptionalInt` from the `GameRegistry` once and if it exists use it here.
No need for `isFuelHandled`.